server/lib/manifest.py

```diff
@@ -0,0 +1,389 @@
+        # Add any external datasets to the manifest
+        self.add_dataset_records(user)
+        self.add_system_files()
+
+    publishers = {
+        "DataONE":
+            {
+                "@id": "https://www.dataone.org/",
+                "@type": "Organization",
+                "legalName": "DataONE",
+                "Description": "A federated data network allowing access to science data"
+            },
+        "Globus":
+            {
+                "@id": "https://www.materialsdatafacility.org/",
+                "@type": "Organization",
+                "legalName": "Materials Data Facility",
+                "Description": "A simple way to publish, discover, and access materials datasets"
+            }
+    }
+
+    def create_basic_attributes(self, tale):
+        """
+        Returns a portion of basic attributes in the manifest
+        :param tale: A Tale that is being described
+        :return: Basic information about the Tale
+        """
+
+        return {
+            "@id": 'https://data.wholetale.org/api/v1/tale/' + str(tale['_id']),
+            "createdOn": str(tale['created']),
+            "schema:name": tale['title'],
+            "schema:description": tale.get('description', str()),
+            "schema:category": tale['category'],
+            "schema:identifier": str(tale['_id']),
+            "schema:version": tale['format'],
+            "schema:image": tale['illustration'],
+            "aggregates": list(),
+            "Datasets": list()
+        }
+
+    def add_tale_creator(self, tale, user):
+        """
+        Adds basic information about the Tale author
+        :param tale: The Tale being described
+        :param user: The user whose information is being used
+        """
+
+        tale_user = self.userModel.load(tale['creatorId'], user=user)
+        self.manifest['createdBy'] = {
+            "@id": tale['authors'],
+            "@type": "schema:Person",
+            "schema:givenName": tale_user.get('firstName', ''),
+            "schema:familyName": tale_user.get('lastName', ''),
+            "schema:email": tale_user.get('email', '')
+        }
+
+    def create_context(self):
+        """
+        Creates the manifest namespace. When a new vocabulary is used, it shoud
+        get added here.
+        :return: A structure defining the used vocabularies
+        """
+        return {
+            "@context": [
+                "https://w3id.org/bundle/context",
+                {"schema": "http://schema.org/"}
+            ]
+        }
+
+    def create_dataset_record(self, user, folder_id):
+        """
+        Creates a record that describes a Dataset
+        :param user: The user
+        :param folder_id: Folder that represents a dataset
+        :return: Dictionary that describes a dataset
+        """
+        folder = self.folderModel.load(folder_id,
+                                       user=user,
+                                       force=True)
```

Comment:
I think we should respect ACLs here. User who doesn't have access to folder shouldn't be able to create manifest for it. `(force=False)`. I think there are more places throughout this class where that applies.

Reply:
In the `Fix permission issues` commit I changed the ACL to READ and removed the `force` kwargs from model and item loads.